presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/PredicatePushDown.java

```diff
@@ -697,19 +725,19 @@ private JoinNode tryNormalizeToOuterToInnerJoin(JoinNode node, Expression inheri
                     return node;
                 }
                 if (canConvertToLeftJoin && canConvertToRightJoin) {
-                    return new JoinNode(node.getId(), INNER, node.getLeft(), node.getRight(), node.getCriteria(), node.getFilter(), node.getLeftHashSymbol(), node.getRightHashSymbol());
+                    return new JoinNode(node.getId(), INNER, node.getLeft(), node.getRight(), node.getCriteria(), node.getOutputSymbols(), node.getFilter(), node.getLeftHashSymbol(), node.getRightHashSymbol());
```

Comment:
It's hard to do that here. This method returns JoinNode, and the caller expects it to be a JoinNode. 

I don't think it can ever be a cross join, though. The original Join node has to have a criteria or filter if it's a FULL OUTER JOIN, and this method does not change or remove them